src/models/PrimaryAuth.js

```diff
@@ -161,12 +159,16 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
       }
     },
 
-    doPrimaryAuth: function (authClient, deviceFingerprintEnabled, signInArgs, func) {
+    doPrimaryAuth: function (authClient, signInArgs, func) {
+      var deviceFingerprintEnabled = this.settings.get('features.deviceFingerprinting');
       // Add the custom header for fingerprint if needed, and then remove it afterwards
       // Since we only need to send it for primary auth
```

Comment:
I see that we are not removing the header after primary auth...  As far as I remember `authClient.options.headers` persist after this call, so we will be sending an empty header in all the requests after this one. Thoughts? Might be a good idea since we only need it for primaryAuth when we capture the username pattern.

Reply:
Good catch ! Fixed now.